src/biconnectivity13/biconnect.jl

```diff
@@ -84,16 +27,72 @@ julia> biconnected_components(cycle_graph(5))
 """
 function biconnected_components end
 @traitfn function biconnected_components(g::::(!IsDirected))
-    state = Biconnections(g)
-    for u in vertices(g)
-        if state.depth[u] == 0
-            visit!(g, state, u, u)
-        end
+        n = nv(g)
+        S = Array{Tuple{Int, Int}, 1}(undef,n)
+        E = Edge{eltype(g)}
+        edge_st = Vector{E}()
+        components = Vector{Vector{E}}()
+        idx = zero(Int)      # preorder counter
+        low = zeros(Int, nv(g))
```

Comment:
you've previously defined `n` (suggest renaming it to `nvg` for clarity). Use that now.

Also, are you sure these should be `Int`s and not `eltype(g)`?